File: client/src/models/user/UserDetailViewModel.ts
import { z } from 'zod';
import AddressViewModel from '../AddressViewModel';

const UserDetailViewModel = z.object({
  id: z.string(),
  name: z.string(),
  avatarImageUrl: z.string().nullable(),
  libraryId: z.string().nullable(),
  roles: z.string().array(),
  address: AddressViewModel.nullable(),
  aboutMe: z.string().nullable()
});

export default UserDetailViewModel;
export type UserDetailViewModelType = z.infer<typeof UserDetailViewModel>;

File: Server/src/Server.Application/ViewModels/UserDetailViewModel.cs
namespace Server.Application.ViewModels;

public class UserDetailViewModel
{
    public Guid Id { get; set; }
    public string? Name { get;  set; }
    public string? AvatarImageUrl { get; set; }
    public Guid? LibraryId { get; set; }
    public List<string> Roles { get; set; } = new();
    public AddressViewModel? Address { get; set; }
}